<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="26" passed="10" failed="16" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2025-10-06T18:04:39 IST" name="U-Ask Chatbot Test Suite" finished-at="2025-10-06T18:14:39 IST" duration-ms="599610">
    <groups>
    </groups>
    <test started-at="2025-10-06T18:04:39 IST" name="English Language Tests" finished-at="2025-10-06T18:09:33 IST" duration-ms="294094">
      <class name="tests.ChatbotUITests">
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:04:39 IST" name="setUp" finished-at="2025-10-06T18:04:42 IST" duration-ms="2963" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testChatInterfaceLoads()[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:04:42 IST" name="testChatInterfaceLoads" finished-at="2025-10-06T18:04:52 IST" duration-ms="10258" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Send button should be present expected [false] but found [true]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Send button should be present expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertFalse(Assert.java:79)
at tests.ChatbotUITests.testChatInterfaceLoads(ChatbotUITests.java:34)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testChatInterfaceLoads -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:04:52 IST" name="tearDown" finished-at="2025-10-06T18:04:52 IST" duration-ms="98" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:04:52 IST" name="setUp" finished-at="2025-10-06T18:04:54 IST" duration-ms="1613" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testInputClearAfterSend()[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:04:54 IST" name="testInputClearAfterSend" finished-at="2025-10-06T18:04:56 IST" duration-ms="2361" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInputClearAfterSend -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:04:56 IST" name="tearDown" finished-at="2025-10-06T18:04:57 IST" duration-ms="95" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:04:57 IST" name="setUp" finished-at="2025-10-06T18:04:58 IST" duration-ms="1682" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testMessageSendFunctionality()[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:04:58 IST" name="testMessageSendFunctionality" finished-at="2025-10-06T18:05:40 IST" duration-ms="41544" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testMessageSendFunctionality -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:05:40 IST" name="tearDown" finished-at="2025-10-06T18:05:40 IST" duration-ms="121" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:05:40 IST" name="setUp" finished-at="2025-10-06T18:05:41 IST" duration-ms="1544" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testResponseRendering()[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:05:41 IST" name="testResponseRendering" finished-at="2025-10-06T18:05:47 IST" duration-ms="5488" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testResponseRendering -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:05:47 IST" name="tearDown" finished-at="2025-10-06T18:05:47 IST" duration-ms="95" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:05:47 IST" name="setUp" finished-at="2025-10-06T18:05:49 IST" duration-ms="1593" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testScrollFunctionality()[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:05:49 IST" name="testScrollFunctionality" finished-at="2025-10-06T18:06:26 IST" duration-ms="37687" status="FAIL">
          <exception class="org.openqa.selenium.TimeoutException">
            <message>
              <![CDATA[Expected condition failed: waiting for element to be clickable: By.cssSelector: button[data-testid="send-button"] (tried for 30 second(s) with 500 milliseconds interval)
Build info: version: '4.15.0', revision: '1d14b5521b'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.3', java.version: '25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 140.0.7339.214, chrome: {chromedriverVersion: 140.0.7339.207 (17230b545fd..., userDataDir: /var/folders/dc/sh_tnvj14q9...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54467}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54467/devtoo..., se:cdpVersion: 140.0.7339.214, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 8bf789a41dcad2dc98fd4f3128d0e721]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.openqa.selenium.TimeoutException: Expected condition failed: waiting for element to be clickable: By.cssSelector: button[data-testid="send-button"] (tried for 30 second(s) with 500 milliseconds interval)
Build info: version: '4.15.0', revision: '1d14b5521b'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.3', java.version: '25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 140.0.7339.214, chrome: {chromedriverVersion: 140.0.7339.207 (17230b545fd..., userDataDir: /var/folders/dc/sh_tnvj14q9...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54467}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54467/devtoo..., se:cdpVersion: 140.0.7339.214, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 8bf789a41dcad2dc98fd4f3128d0e721
at org.openqa.selenium.support.ui.WebDriverWait.timeoutException(WebDriverWait.java:84)
at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:229)
at tests.ChatbotUITests.testScrollFunctionality(ChatbotUITests.java:130)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- org.openqa.selenium.TimeoutException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testScrollFunctionality -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.ChatbotUITests@43ee72e6]" started-at="2025-10-06T18:06:26 IST" name="tearDown" finished-at="2025-10-06T18:06:26 IST" duration-ms="151" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- tests.ChatbotUITests -->
      <class name="tests.SecurityTests">
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:27 IST" name="setUp" finished-at="2025-10-06T18:08:28 IST" duration-ms="1512" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testLongInputHandling()[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:28 IST" name="testLongInputHandling" finished-at="2025-10-06T18:08:38 IST" duration-ms="9271" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Should handle long input gracefully and provide a response expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Should handle long input gracefully and provide a response expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at tests.SecurityTests.testLongInputHandling(SecurityTests.java:195)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLongInputHandling -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:38 IST" name="tearDown" finished-at="2025-10-06T18:08:38 IST" duration-ms="99" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:38 IST" name="setUp" finished-at="2025-10-06T18:08:39 IST" duration-ms="1475" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testPromptInjectionHandling()[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:39 IST" name="testPromptInjectionHandling" finished-at="2025-10-06T18:08:45 IST" duration-ms="5469" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testPromptInjectionHandling -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:45 IST" name="tearDown" finished-at="2025-10-06T18:08:45 IST" duration-ms="97" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:45 IST" name="setUp" finished-at="2025-10-06T18:08:46 IST" duration-ms="1602" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testScriptInjectionHandling()[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:46 IST" name="testScriptInjectionHandling" finished-at="2025-10-06T18:08:52 IST" duration-ms="5483" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Should respond to the legitimate part of the query expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Should respond to the legitimate part of the query expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at tests.SecurityTests.testScriptInjectionHandling(SecurityTests.java:65)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testScriptInjectionHandling -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:52 IST" name="tearDown" finished-at="2025-10-06T18:08:52 IST" duration-ms="99" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:52 IST" name="setUp" finished-at="2025-10-06T18:08:54 IST" duration-ms="1691" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testSpecialCharactersHandling()[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:08:54 IST" name="testSpecialCharactersHandling" finished-at="2025-10-06T18:09:33 IST" duration-ms="39548" status="FAIL">
          <exception class="org.openqa.selenium.TimeoutException">
            <message>
              <![CDATA[Expected condition failed: waiting for element to be clickable: By.cssSelector: button[data-testid="send-button"] (tried for 30 second(s) with 500 milliseconds interval)
Build info: version: '4.15.0', revision: '1d14b5521b'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.3', java.version: '25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 140.0.7339.214, chrome: {chromedriverVersion: 140.0.7339.207 (17230b545fd..., userDataDir: /var/folders/dc/sh_tnvj14q9...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54786}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54786/devtoo..., se:cdpVersion: 140.0.7339.214, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 3ac90df54751c07388c66817a5f34bd0]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.openqa.selenium.TimeoutException: Expected condition failed: waiting for element to be clickable: By.cssSelector: button[data-testid="send-button"] (tried for 30 second(s) with 500 milliseconds interval)
Build info: version: '4.15.0', revision: '1d14b5521b'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.3', java.version: '25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 140.0.7339.214, chrome: {chromedriverVersion: 140.0.7339.207 (17230b545fd..., userDataDir: /var/folders/dc/sh_tnvj14q9...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54786}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54786/devtoo..., se:cdpVersion: 140.0.7339.214, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 3ac90df54751c07388c66817a5f34bd0
at org.openqa.selenium.support.ui.WebDriverWait.timeoutException(WebDriverWait.java:84)
at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:229)
at tests.SecurityTests.testSpecialCharactersHandling(SecurityTests.java:128)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- org.openqa.selenium.TimeoutException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testSpecialCharactersHandling -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.SecurityTests@18ce0030]" started-at="2025-10-06T18:09:33 IST" name="tearDown" finished-at="2025-10-06T18:09:33 IST" duration-ms="127" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- tests.SecurityTests -->
      <class name="tests.AIResponseTests">
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:06:26 IST" name="setUp" finished-at="2025-10-06T18:06:28 IST" duration-ms="1599" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testCommonQueriesResponseQuality()[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:06:28 IST" name="testCommonQueriesResponseQuality" finished-at="2025-10-06T18:07:05 IST" duration-ms="36608" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Response for 'What is artificial intelligence?' should not be hallucinated expected [false] but found [true]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Response for 'What is artificial intelligence?' should not be hallucinated expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertFalse(Assert.java:79)
at tests.AIResponseTests.testCommonQueriesResponseQuality(AIResponseTests.java:61)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCommonQueriesResponseQuality -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:07:05 IST" name="tearDown" finished-at="2025-10-06T18:07:05 IST" duration-ms="134" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:07:05 IST" name="setUp" finished-at="2025-10-06T18:07:09 IST" duration-ms="4241" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testFallbackMessages()[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:07:09 IST" name="testFallbackMessages" finished-at="2025-10-06T18:07:45 IST" duration-ms="36422" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testFallbackMessages -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:07:45 IST" name="tearDown" finished-at="2025-10-06T18:07:46 IST" duration-ms="111" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:07:46 IST" name="setUp" finished-at="2025-10-06T18:07:47 IST" duration-ms="1839" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testResponseFormatting()[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:07:47 IST" name="testResponseFormatting" finished-at="2025-10-06T18:08:24 IST" duration-ms="36625" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Response should be substantial in length expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Response should be substantial in length expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at tests.AIResponseTests.testResponseFormatting(AIResponseTests.java:98)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testResponseFormatting -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:08:24 IST" name="tearDown" finished-at="2025-10-06T18:08:24 IST" duration-ms="181" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:08:24 IST" name="setUp" finished-at="2025-10-06T18:08:26 IST" duration-ms="1715" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[en]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testResponseTime()[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:08:26 IST" name="testResponseTime" finished-at="2025-10-06T18:08:27 IST" duration-ms="719" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testResponseTime -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.AIResponseTests@56a6d5a6]" started-at="2025-10-06T18:08:27 IST" name="tearDown" finished-at="2025-10-06T18:08:27 IST" duration-ms="96" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- tests.AIResponseTests -->
    </test> <!-- English Language Tests -->
    <test started-at="2025-10-06T18:09:33 IST" name="Arabic Language Tests" finished-at="2025-10-06T18:14:39 IST" duration-ms="305357">
      <class name="tests.ChatbotUITests">
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:09:33 IST" name="setUp" finished-at="2025-10-06T18:09:35 IST" duration-ms="1655" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testChatInterfaceLoads()[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:09:35 IST" name="testChatInterfaceLoads" finished-at="2025-10-06T18:09:45 IST" duration-ms="10225" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Send button should be present expected [false] but found [true]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Send button should be present expected [false] but found [true]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertFalse(Assert.java:79)
at tests.ChatbotUITests.testChatInterfaceLoads(ChatbotUITests.java:34)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testChatInterfaceLoads -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:09:45 IST" name="tearDown" finished-at="2025-10-06T18:09:45 IST" duration-ms="101" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:09:45 IST" name="setUp" finished-at="2025-10-06T18:09:47 IST" duration-ms="1711" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testInputClearAfterSend()[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:09:47 IST" name="testInputClearAfterSend" finished-at="2025-10-06T18:09:49 IST" duration-ms="2409" status="FAIL">
          <exception class="org.openqa.selenium.NoSuchSessionException">
            <message>
              <![CDATA[invalid session id: session deleted as the browser has closed the connection
from disconnected: not connected to DevTools
  (Session info: chrome=140.0.7339.214)
Build info: version: '4.15.0', revision: '1d14b5521b'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.3', java.version: '25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [4a1776d099f4bbef29fd8dff650f8537, findElement {using=css selector, value=#prompt-textarea}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 140.0.7339.214, chrome: {chromedriverVersion: 140.0.7339.207 (17230b545fd..., userDataDir: /var/folders/dc/sh_tnvj14q9...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54867}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54867/devtoo..., se:cdpVersion: 140.0.7339.214, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 4a1776d099f4bbef29fd8dff650f8537]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.openqa.selenium.NoSuchSessionException: invalid session id: session deleted as the browser has closed the connection
from disconnected: not connected to DevTools
  (Session info: chrome=140.0.7339.214)
Build info: version: '4.15.0', revision: '1d14b5521b'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.3', java.version: '25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [4a1776d099f4bbef29fd8dff650f8537, findElement {using=css selector, value=#prompt-textarea}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 140.0.7339.214, chrome: {chromedriverVersion: 140.0.7339.207 (17230b545fd..., userDataDir: /var/folders/dc/sh_tnvj14q9...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54867}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54867/devtoo..., se:cdpVersion: 140.0.7339.214, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 4a1776d099f4bbef29fd8dff650f8537
at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:483)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:200)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:133)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:52)
at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:191)
at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:200)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:175)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:607)
at org.openqa.selenium.remote.ElementLocation$ElementFinder$2.findElement(ElementLocation.java:165)
at org.openqa.selenium.remote.ElementLocation.findElement(ElementLocation.java:59)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:364)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:358)
at tests.ChatbotUITests.testInputClearAfterSend(ChatbotUITests.java:116)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- org.openqa.selenium.NoSuchSessionException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testInputClearAfterSend -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:09:49 IST" name="tearDown" finished-at="2025-10-06T18:09:50 IST" duration-ms="50" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:09:50 IST" name="setUp" finished-at="2025-10-06T18:09:51 IST" duration-ms="1484" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testMessageSendFunctionality()[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:09:51 IST" name="testMessageSendFunctionality" finished-at="2025-10-06T18:10:37 IST" duration-ms="45689" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testMessageSendFunctionality -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:10:37 IST" name="tearDown" finished-at="2025-10-06T18:10:37 IST" duration-ms="111" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:10:37 IST" name="setUp" finished-at="2025-10-06T18:10:38 IST" duration-ms="1588" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testResponseRendering()[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:10:38 IST" name="testResponseRendering" finished-at="2025-10-06T18:10:44 IST" duration-ms="5398" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testResponseRendering -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:10:44 IST" name="tearDown" finished-at="2025-10-06T18:10:44 IST" duration-ms="124" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:10:44 IST" name="setUp" finished-at="2025-10-06T18:10:47 IST" duration-ms="2990" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testScrollFunctionality()[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:10:47 IST" name="testScrollFunctionality" finished-at="2025-10-06T18:11:24 IST" duration-ms="37573" status="FAIL">
          <exception class="org.openqa.selenium.TimeoutException">
            <message>
              <![CDATA[Expected condition failed: waiting for element to be clickable: By.cssSelector: button[data-testid="send-button"] (tried for 30 second(s) with 500 milliseconds interval)
Build info: version: '4.15.0', revision: '1d14b5521b'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.3', java.version: '25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 140.0.7339.214, chrome: {chromedriverVersion: 140.0.7339.207 (17230b545fd..., userDataDir: /var/folders/dc/sh_tnvj14q9...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54982}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54982/devtoo..., se:cdpVersion: 140.0.7339.214, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: c3ece1335252104378058dff3d7e64f9]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.openqa.selenium.TimeoutException: Expected condition failed: waiting for element to be clickable: By.cssSelector: button[data-testid="send-button"] (tried for 30 second(s) with 500 milliseconds interval)
Build info: version: '4.15.0', revision: '1d14b5521b'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.3', java.version: '25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 140.0.7339.214, chrome: {chromedriverVersion: 140.0.7339.207 (17230b545fd..., userDataDir: /var/folders/dc/sh_tnvj14q9...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:54982}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:54982/devtoo..., se:cdpVersion: 140.0.7339.214, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: c3ece1335252104378058dff3d7e64f9
at org.openqa.selenium.support.ui.WebDriverWait.timeoutException(WebDriverWait.java:84)
at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:229)
at tests.ChatbotUITests.testScrollFunctionality(ChatbotUITests.java:130)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- org.openqa.selenium.TimeoutException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testScrollFunctionality -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.ChatbotUITests@1ed1993a]" started-at="2025-10-06T18:11:24 IST" name="tearDown" finished-at="2025-10-06T18:11:25 IST" duration-ms="133" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- tests.ChatbotUITests -->
      <class name="tests.SecurityTests">
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:24 IST" name="setUp" finished-at="2025-10-06T18:13:25 IST" duration-ms="1607" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testLongInputHandling()[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:25 IST" name="testLongInputHandling" finished-at="2025-10-06T18:13:34 IST" duration-ms="9285" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testLongInputHandling -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:34 IST" name="tearDown" finished-at="2025-10-06T18:13:35 IST" duration-ms="109" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:35 IST" name="setUp" finished-at="2025-10-06T18:13:36 IST" duration-ms="1476" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testPromptInjectionHandling()[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:36 IST" name="testPromptInjectionHandling" finished-at="2025-10-06T18:13:42 IST" duration-ms="5545" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Should provide a reasonable response to prompt injection attempt expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Should provide a reasonable response to prompt injection attempt expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at tests.SecurityTests.testPromptInjectionHandling(SecurityTests.java:109)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testPromptInjectionHandling -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:42 IST" name="tearDown" finished-at="2025-10-06T18:13:42 IST" duration-ms="99" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:42 IST" name="setUp" finished-at="2025-10-06T18:13:43 IST" duration-ms="1805" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testScriptInjectionHandling()[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:43 IST" name="testScriptInjectionHandling" finished-at="2025-10-06T18:13:49 IST" duration-ms="5471" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Should respond to the legitimate part of the query expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Should respond to the legitimate part of the query expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at tests.SecurityTests.testScriptInjectionHandling(SecurityTests.java:65)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testScriptInjectionHandling -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:49 IST" name="tearDown" finished-at="2025-10-06T18:13:49 IST" duration-ms="101" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:49 IST" name="setUp" finished-at="2025-10-06T18:13:50 IST" duration-ms="1120" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testSpecialCharactersHandling()[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:13:50 IST" name="testSpecialCharactersHandling" finished-at="2025-10-06T18:14:39 IST" duration-ms="48390" status="FAIL">
          <exception class="org.openqa.selenium.TimeoutException">
            <message>
              <![CDATA[Expected condition failed: waiting for element to be clickable: By.cssSelector: button[data-testid="send-button"] (tried for 30 second(s) with 500 milliseconds interval)
Build info: version: '4.15.0', revision: '1d14b5521b'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.3', java.version: '25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 140.0.7339.214, chrome: {chromedriverVersion: 140.0.7339.207 (17230b545fd..., userDataDir: /var/folders/dc/sh_tnvj14q9...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:55288}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:55288/devtoo..., se:cdpVersion: 140.0.7339.214, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 6898f0e910636a7981028a49e0fa4067]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.openqa.selenium.TimeoutException: Expected condition failed: waiting for element to be clickable: By.cssSelector: button[data-testid="send-button"] (tried for 30 second(s) with 500 milliseconds interval)
Build info: version: '4.15.0', revision: '1d14b5521b'
System info: os.name: 'Mac OS X', os.arch: 'aarch64', os.version: '14.3', java.version: '25'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 140.0.7339.214, chrome: {chromedriverVersion: 140.0.7339.207 (17230b545fd..., userDataDir: /var/folders/dc/sh_tnvj14q9...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:55288}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: mac, proxy: Proxy(), se:cdp: ws://localhost:55288/devtoo..., se:cdpVersion: 140.0.7339.214, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: 6898f0e910636a7981028a49e0fa4067
at org.openqa.selenium.support.ui.WebDriverWait.timeoutException(WebDriverWait.java:84)
at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:229)
at tests.SecurityTests.testSpecialCharactersHandling(SecurityTests.java:128)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- org.openqa.selenium.TimeoutException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testSpecialCharactersHandling -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.SecurityTests@794cb805]" started-at="2025-10-06T18:14:39 IST" name="tearDown" finished-at="2025-10-06T18:14:39 IST" duration-ms="130" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- tests.SecurityTests -->
      <class name="tests.AIResponseTests">
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:11:25 IST" name="setUp" finished-at="2025-10-06T18:11:26 IST" duration-ms="1749" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testCommonQueriesResponseQuality()[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:11:26 IST" name="testCommonQueriesResponseQuality" finished-at="2025-10-06T18:12:03 IST" duration-ms="36732" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Response for 'ما هو الذكاء الاصطناعي؟' should be valid and contain relevant keywords expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Response for 'ما هو الذكاء الاصطناعي؟' should be valid and contain relevant keywords expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at tests.AIResponseTests.testCommonQueriesResponseQuality(AIResponseTests.java:56)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testCommonQueriesResponseQuality -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:12:03 IST" name="tearDown" finished-at="2025-10-06T18:12:03 IST" duration-ms="124" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:12:03 IST" name="setUp" finished-at="2025-10-06T18:12:05 IST" duration-ms="1741" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testFallbackMessages()[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:12:05 IST" name="testFallbackMessages" finished-at="2025-10-06T18:12:42 IST" duration-ms="36647" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Should provide a reasonable response even for unclear input expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Should provide a reasonable response even for unclear input expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at tests.AIResponseTests.testFallbackMessages(AIResponseTests.java:151)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testFallbackMessages -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:12:42 IST" name="tearDown" finished-at="2025-10-06T18:12:42 IST" duration-ms="125" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:12:42 IST" name="setUp" finished-at="2025-10-06T18:12:44 IST" duration-ms="2709" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testResponseFormatting()[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:12:44 IST" name="testResponseFormatting" finished-at="2025-10-06T18:13:21 IST" duration-ms="36450" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Response should be substantial in length expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Response should be substantial in length expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at tests.AIResponseTests.testResponseFormatting(AIResponseTests.java:98)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
at java.base/java.lang.reflect.Method.invoke(Method.java:565)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
at java.base/java.lang.Thread.run(Thread.java:1474)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testResponseFormatting -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:13:21 IST" name="tearDown" finished-at="2025-10-06T18:13:21 IST" duration-ms="122" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="setUp(java.lang.String)[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:13:21 IST" name="setUp" finished-at="2025-10-06T18:13:23 IST" duration-ms="1794" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ar]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method signature="testResponseTime()[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:13:23 IST" name="testResponseTime" finished-at="2025-10-06T18:13:23 IST" duration-ms="651" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testResponseTime -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:tests.AIResponseTests@1f3f4916]" started-at="2025-10-06T18:13:23 IST" name="tearDown" finished-at="2025-10-06T18:13:24 IST" duration-ms="106" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- tests.AIResponseTests -->
    </test> <!-- Arabic Language Tests -->
  </suite> <!-- U-Ask Chatbot Test Suite -->
</testng-results>
